calculator.rb

```diff
@@ -0,0 +1,134 @@
+# Methods
+def add (num1, num2)
+  return num1 + num2
+end
+
+def subtract (num1, num2)
+  return num1 - num2
+end
+
+def multiply (num1, num2)
+  return num1 * num2
+end
+
+def divide (num1, num2)
+  return num1 / num2
+end 
+
+def exponent (num1, num2)
+  return num1 ** num2
+end 
+
+def modulo (num1, num2)
+  return num1 % num2
+end 
+
+def num_check (num)
+  until num.to_f.to_s == num.to_s || num.to_i.to_s == num.to_s
+    puts "That number is invalid. Please try again: "
+    num = gets.chomp 
+  end
+  return num.to_f
+end 
+
+def float_int_check (num)
+  if num == num.to_i
+    return num.to_i
+  else return num
+  end
+end 
+
+# Intro and get user input
+puts "Welcome to your handy dandy calculator!"
+print "What would you like to do? 
+Add (+), Subtract (-), Multiply (*), Divide (/), Exponent (^), or Modulo (%)? "
+oper = gets.chomp.upcase
+
+# Check for valid user input by checking it against array of valid operations/symbols
+operations = ["ADD", "+", "SUBTRACT", "-", "MULTIPLY", "*", "X", "DIVIDE", "/", 
+  "EXPONENT", "^", "**", "MODULO", "MODULUS", "REMAINDER", "%"]
+  until operations.include?(oper)
+    print "That operation is invalid. Please try again: "
+    oper = gets.chomp.upcase
+  end 
+  
+  # Addition
+  if oper == "ADD" || oper == "+"
```

Comment:
I'm surprised you didn't opt for a `case` `when` here, but it works all the same!